### ISSUE TITLE
Error updating conditional access "'GuestsOrExternalUsers' cannot be combined with guestOrExternalUsers"

### ISSUE THREAD

**Issue by DevopsMercenary** [hashicorp/terraform-provider-azuread#1196]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureAD Provider) Version

$ terraform -v
Terraform v1.5.6
on darwin_arm64
+ provider registry.terraform.io/hashicorp/azuread v2.42.0
+ provider registry.terraform.io/hashicorp/azurerm v3.73.0

### Affected Resource(s)

* `azuread_conditional_access_policy`

### Terraform Configuration Files

```hcl
resource "azuread_conditional_access_policy" "require-multifactor-authentication-for-guest-access" {

  display_name = "Require multifactor authentication for guest access"
  
  state        = "enabled"

  conditions {
    client_app_types = ["all"]
    applications {
      included_applications = ["All"]
    }
    users {
      included_users = ["GuestsOrExternalUsers"]
    }
  }
  grant_controls {
    operator          = "OR"
    built_in_controls = ["mfa"]
  }
}
```

### Debug Output

https://gist.github.com/DevopsMercenary/4b3fe844f911f0883cfb169c8acf625b

### Actual Behavior

```
  # azuread_conditional_access_policy.require-multifactor-authentication-for-guest-access will be updated in-place
  ~ resource "azuread_conditional_access_policy" "require-multifactor-authentication-for-guest-access" {
        id           = "ea46fc09-4bca-489e-b3b4-ad1f2afacab3"
        # (2 unchanged attributes hidden)

      ~ conditions {
            # (4 unchanged attributes hidden)

          ~ users {
              ~ included_users  = [
                  + "GuestsOrExternalUsers",
                ]
                # (5 unchanged attributes hidden)
            }

            # (1 unchanged block hidden)
        }

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value:
azuread_conditional_access_policy.require-multifactor-authentication-for-guest-access: Modifying... [id=ea46fc09-4bca-489e-b3b4-ad1f2afacab3]
╷
│ Error: Could not update conditional access policy with ID: "ea46fc09-4bca-489e-b3b4-ad1f2afacab3"
│
│   with azuread_conditional_access_policy.require-multifactor-authentication-for-guest-access,
│   on require-multifactor-authentication-for-guest-access.tf line 5, in resource "azuread_conditional_access_policy" "require-multifactor-authentication-for-guest-access":
│    5: resource "azuread_conditional_access_policy" "require-multifactor-authentication-for-guest-access" {
│
│ ConditionalAccessPoliciesClient.BaseClient.Patch(): unexpected status 400 with OData error: BadRequest: 1119: Users assignment value 'None', 'All' or 'GuestsOrExternalUsers' cannot be combined with guestOrExternalUsers. Please remove one of the
│ assignment and try again. For examples, please see API documentation at https://docs.microsoft.com/en-us/graph/api/conditionalaccesspolicy-update?view=graph-rest-1.0.
╵
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by manicminer** [comment]
@DevopsMercenary Many thanks for reporting this error. At first glance this looks like a potential breaking change in the API, however using your exact configuration I was not able to reproduce the error. Perhaps it was a transient issue, are you still experiencing this? Thanks!

**Comment by DevopsMercenary** [comment]
Yes, still having an issue with updating this conditional access policy.

I updated everything before I checked it today...

```
$ tf --version
Terraform v1.5.7
on darwin_arm64
+ provider registry.terraform.io/hashicorp/azuread v2.43.0
+ provider registry.terraform.io/hashicorp/azurerm v3.74.0
```

Error remains the same though

```

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # azuread_conditional_access_policy.require-multifactor-authentication-for-guest-access will be updated in-place
  ~ resource "azuread_conditional_access_policy" "require-multifactor-authentication-for-guest-access" {
        id           = "ea46fc09-4bca-489e-b3b4-ad1f2afacab3"
        # (2 unchanged attributes hidden)

      ~ conditions {
            # (4 unchanged attributes hidden)

          ~ users {
              ~ included_users  = [
                  + "GuestsOrExternalUsers",
                ]
                # (5 unchanged attributes hidden)
            }

            # (1 unchanged block hidden)
        }

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

azuread_conditional_access_policy.require-multifactor-authentication-for-guest-access: Modifying... [id=ea46fc09-4bca-489e-b3b4-ad1f2afacab3]
╷
│ Error: Could not update conditional access policy with ID: "ea46fc09-4bca-489e-b3b4-ad1f2afacab3"
│
│   with azuread_conditional_access_policy.require-multifactor-authentication-for-guest-access,
│   on require-multifactor-authentication-for-guest-access.tf line 5, in resource "azuread_conditional_access_policy" "require-multifactor-authentication-for-guest-access":
│    5: resource "azuread_conditional_access_policy" "require-multifactor-authentication-for-guest-access" {
│
│ ConditionalAccessPoliciesClient.BaseClient.Patch(): unexpected status 400 with OData error: BadRequest: 1119: Users assignment value 'None', 'All' or 'GuestsOrExternalUsers' cannot be combined with guestOrExternalUsers. Please remove one of the
│ assignment and try again. For examples, please see API documentation at https://docs.microsoft.com/en-us/graph/api/conditionalaccesspolicy-update?view=graph-rest-1.0.
╵
```

**Comment by manicminer** [comment]
@DevopsMercenary Are you able to post the existing policy (Terraform config and/or the actual full policy)? Is this perhaps a policy that was initially created outside of Terraform? Thanks!

**Comment by DevopsMercenary** [comment]
Yes,  There is a chance ( a good one ) that I imported it after using the template.

Here's the JSON...

```
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#identity/conditionalAccess/policies/$entity",
    "@microsoft.graph.tips": "Use $select to choose only the properties your app needs, as this can lead to performance improvements. For example: GET identity/conditionalAccess/policies('<ea46fc09-4bca-489e-b3b4-ad1f2afacab3>')?$select=conditions,createdDateTime",
    "id": "ea46fc09-4bca-489e-b3b4-ad1f2afacab3",
    "templateId": null,
    "displayName": "Require multifactor authentication for guest access",
    "createdDateTime": "2023-09-01T19:27:25.521493Z",
    "modifiedDateTime": "2023-09-04T17:14:14.1246749Z",
    "state": "enabled",
    "sessionControls": null,
    "conditions": {
        "userRiskLevels": [],
        "signInRiskLevels": [],
        "clientAppTypes": [
            "all"
        ],
        "servicePrincipalRiskLevels": [],
        "platforms": null,
        "locations": null,
        "devices": null,
        "clientApplications": null,
        "applications": {
            "includeApplications": [
                "All"
            ],
            "excludeApplications": [],
            "includeUserActions": [],
            "includeAuthenticationContextClassReferences": [],
            "applicationFilter": null
        },
        "users": {
            "includeUsers": [],
            "excludeUsers": [],
            "includeGroups": [],
            "excludeGroups": [],
            "includeRoles": [],
            "excludeRoles": [],
            "excludeGuestsOrExternalUsers": null,
            "includeGuestsOrExternalUsers": {
                "guestOrExternalUserTypes": "internalGuest,b2bCollaborationGuest,b2bCollaborationMember,b2bDirectConnectUser,otherExternalUser,serviceProvider",
                "externalTenants": {
                    "@odata.type": "#microsoft.graph.conditionalAccessAllExternalTenants",
                    "membershipKind": "all"
                }
            }
        }
    },
    "grantControls": {
        "operator": "OR",
        "builtInControls": [
            "mfa"
        ],
        "customAuthenticationFactors": [],
        "termsOfUse": [],
        "authenticationStrength@odata.context": "https://graph.microsoft.com/v1.0/$metadata#identity/conditionalAccess/policies('ea46fc09-4bca-489e-b3b4-ad1f2afacab3')/grantControls/authenticationStrength/$entity",
        "authenticationStrength": null
    }
}
```

**Comment by manicminer** [comment]
@DevopsMercenary This is due to the `includeGuestsOrExternalUsers` condition field being set on the policy you're importing, a property which we don't yet support which is why the provider is not unsetting it to accommodate your new `included_users` property.

This should be resolved by #1222.
